src/Microsoft.Framework.PackageManager/Utils/LockFileUtils.cs

```diff
@@ -18,10 +18,11 @@ internal static class LockFileUtils
             IPackage package,
             SHA512 sha512,
             IEnumerable<FrameworkName> frameworks,
-            IPackagePathResolver resolver)
+            IPackagePathResolver resolver,
+            string correctedPackageName = null)
         {
             var lockFileLib = new LockFileLibrary();
-            lockFileLib.Name = package.Id;
+            lockFileLib.Name = correctedPackageName ?? package.Id;
```

Comment:
@davidfowl , I fixed the bug with this minimal change.
`package.Id` is read from nuspec of an installed package and its casing might be wrong. Here we use the package name from runtime dependency graph as corrected package name.